sdk/identity/azure-identity/azure/identity/aio/_credentials/imds.py

```diff
@@ -65,9 +66,9 @@ async def _acquire_token_silently(self, *scopes: str, **kwargs: Any) -> Optional
 
     async def _request_token(self, *scopes: str, **kwargs: Any) -> AccessTokenInfo:
 
-        if within_credential_chain.get() and not self._endpoint_available:
-            # If within a chain (e.g. DefaultAzureCredential), we do a quick check to see if the IMDS endpoint
-            # is available to avoid hanging for a long time if the endpoint isn't available.
+        do_probe = self._enable_imds_probe if self._enable_imds_probe is not None else within_credential_chain.get()
```

Comment:
Should we do probe if it is in a chained cred? I guess no?

Reply:
That's the behavior we previously had, so I think we should maintain that.